### DISCUSSION TITLE
Error keeps appearing after submit

### DISCUSSION THREAD

**Discussion by brandon-9** [livewire/livewire#2296] [1. Help]
I have a single input field, that on submit creates a new model.

```
    protected $rules = [
        'amenityName' => 'required|min:3|string|unique:amenities,name',
    ];

public function submit()
    {

        $this->validate();
        $addAmenity = Amenity::create([
            'name' => $this->amenityName,
        ]);

        $this->resetForm();
    }

    private function resetForm()
    {
        $this->reset('amenityName');
        $this->resetValidation();
    }
```

After Submit(), the input field clears, the new Model is created and added everywhere, however then the error under the input displays: "The amenity name field is required." - and is the input has been reset and is empty.

Why is the error appearing after submit? Shouldn't the Error Bag clear out?

p.s. I also can't create a user on the Livewire forum as it keeps saying "New registrations are not allowed from your IP address (maximum limit reached). Contact a staff member."

**Comment by tpaksu** [comment]
You might have found the solution before, since it's been 3 months after you asked the question, but you need to clear the error bag before validation to get new validation results in the bag:

```php
public function submit()
{
    $this->resetErrorBag();
    $this->validate();
    ...
```

**Reply by brandon-9** [reply]
Thanks for the suggestion! I actually had that later on as well. My issue ended up being that I had a submit button on the form with the type as submit, and therefore livewire was submitting on @click and then the button was natively firing as well, causing the form to submit twice, and the 2nd one always failed validation because the field was empty.
